### PR TITLE
[bitnami/tomcat] monitoring fixes

### PR DESCRIPTION
**Pull request by cache-sk**
Signed-off-by: Peter Kassak <48527944+cache-sk@users.noreply.github.com>

**Description of the change**

Fixes for monitoring / jmx-exporter.
- jmx-exporter container had bad indent
- jmx-config configmap volme was missing
- podmonitor needs some ifs to not add empty {}/[] to final yml

**Benefits**

Tomcat can now be deployed with monitoring

**Possible drawbacks**

None?

**Applicable issues**

None reported so far.

**Additional information**

None.

**Checklist**
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/tomcat/templates/_pod.tpl [hunk 1]
```diff
@@ -178,6 +178,16 @@ volumes:
...
+  {{- if and .Values.metrics.jmx.enabled .Values.metrics.jmx.config (not .Values.metrics.jmx.existingConfigmap) }}
+  - configMap:
+      name: {{ template "tomcat.fullname" . }}-jmx-configuration
+    name: jmx-config
+  {{- end }}
+  {{- if and .Values.metrics.jmx.enabled .Values.metrics.jmx.existingConfigmap }}
+  - configMap:
+      name: {{ .Values.metrics.jmx.existingConfigmap | quote }}
+    name: jmx-config
+  {{- end }}
```
Comment (migruiz4) [inline bitnami/tomcat/templates/_pod.tpl:190]: Could you please use the `tomcat.metrics.jmx.configmapName` helper instead?

```suggestion
  {{- if and .Values.metrics.jmx.enabled (or Values.metrics.jmx.config .Values.metrics.jmx.existingConfigmap) }}
  - configMap:
      name: {{ include "tomcat.metrics.jmx.configmapName" . }}
    name: jmx-config
  {{- end }}

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @cache-sk,

Thank you very much for your contribution!

I have made a small change request below.

In addition, it would be great if you could:
- Indent the `if`/`end` for the `metrics.jxm.enabled` too.
- Remove the duplicated reference to `.Values.sidecars` that is above the jmx-exporter section (the one without `nindent 2`).

File: bitnami/tomcat/templates/_pod.tpl (hunk 1 above)
Comment (cache-sk) [reply bitnami/tomcat/templates/_pod.tpl]: Done.

File: bitnami/tomcat/templates/_pod.tpl [hunk 2]
```diff
@@ -178,6 +175,16 @@ volumes:
...
+  {{- if and .Values.metrics.jmx.enabled .Values.metrics.jmx.config (not .Values.metrics.jmx.existingConfigmap) }}
+  - configMap:
+      name: {{ include "tomcat.metrics.jmx.configmapName" . }}
+    name: jmx-config
+  {{- end }}
+  {{- if and .Values.metrics.jmx.enabled .Values.metrics.jmx.existingConfigmap }}
+  - configMap:
+      name: {{ .Values.metrics.jmx.existingConfigmap | quote }}
+    name: jmx-config
+  {{- end }}
```
Comment (migruiz4) [inline bitnami/tomcat/templates/_pod.tpl:187]: These two can be reduced to a single conditional, the helper `tomcat.metrics.jmx.configmapName` will take care of setting the name of the `existingConfigmap` or the configmap created when providing `metrics.jmx.config`.

```suggestion
  {{- if and .Values.metrics.jmx.enabled (or .Values.metrics.jmx.config .Values.metrics.jmx.existingConfigmap) }}
  - configMap:
      name: {{ include "tomcat.metrics.jmx.configmapName" . }}
    name: jmx-config
  {{- end }}
```

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Thank you very much for applying the suggested changes! Just a minor change request and it should be ready.

File: bitnami/tomcat/templates/_pod.tpl (hunk 2 above)
Comment (cache-sk) [reply bitnami/tomcat/templates/_pod.tpl]: Done.

**Review by migruiz4 — APPROVED** [review]
Thank you very much @cache-sk! Looks good to me!

**Pull request merged**